src/java.base/share/native/libzip/zlib/gzwrite.c

```diff
@@ -449,7 +449,14 @@ int ZEXPORTVA gzvprintf(gzFile file, const char *format, va_list va)
     (void)vsnprintf(next, state->size, format, va);
     len = strlen(next);
 #  else
-    len = vsnprintf(next, state->size, format, va);
+#   ifdef __APPLE__ // ignore format-nonliteral warning on macOS
```

Comment:
Instead of patching 3rd party code to fix a compilation warning, you should disable that warning instead.

In `make/modules/java.base/lib/CoreLibraries.gmk`, add 
```
DISABLED_WARNINGS_clang := format-nonliteral, \
```
as line 138.

Reply:
Thank you for these useful inputs Magnus. I did these changes locally but for some reason this fromat-nonliteral is getting picked up while building that library. I will investigate and see what's going on. Will update the PR once I figure it out.

Reply:
I agree with Magnus and try to avoid changing the imported zlib code.

Reply:
>  I did these changes locally but for some reason this format-nonliteral is not getting picked up while building that library.

Turns out that was slightly inaccurate. What was actually happening was that, that setting you suggested in that build file did indeed work and got picked up. But it ran into an error which looked like:

```
src/java.base/share/native/libzip/zlib/gzwrite.c:452:40: error: format string is not a string literal [-Werror,-Wformat-nonliteral]
    len = vsnprintf(next, state->size, format, va);
                                       ^~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/secure/_stdio.h:75:63: note: expanded from macro 'vsnprintf'
  __builtin___vsnprintf_chk (str, len, 0, __darwin_obsz(str), format, ap)
                                                              ^~~~~~
1 error generated.
make[3]: *** [build/macosx-aarch64-server-release/support/native/java.desktop/libsplashscreen/gzwrite.o] Error 1
```
I hadn't paid close attention initially and upon seeing that error, I thought it was coming while building `LIBZIP`. However, what's actually happening is, it appears that while building `LIBSPLASHSCREEN` is bundled zlib is used then the build process recompiles the bundled zlib, so I had to include that setting even in the `Awt2dLibraries.gmk`.

I have now updated this PR with the changes you suggested and this additional change. Local build now works fine both for bundled and system zlib. I will trigger more extensive tests once these changes look OK.

Reply:
> I agree with Magnus and try to avoid changing the imported zlib code.

We already had modified gzwrite.c in our port so I thought keeping the changes narrowed to apple made sense here.